### PR TITLE
Strip # characters from color props in label config file

### PR DESCRIPTION
**Pull request by luzlab**
This addresses issue #152 . All the unit tests pass.

### CODE REVIEW

File: bin/github-label-sync.js [hunk 1]
```diff
@@ -137,8 +137,11 @@ function merge(files) {
 		});
 	});
 
+	// Make sure leading `#` are stripped from the color property.
 	Object.keys(data).forEach((key) => {
-		labels.push(data[key]);
+		const label = data[key];
+		label.color = label.color.replace('#', '');
```
Comment (JakeChampion) [inline bin/github-label-sync.js:143]: This removes the first instance of `#`, which is differently from removing it from the start of the string.

For example: `'a#b'.replace('#', '')` would return `'ab'`.

Should we change the code to only remove it from the start of the string?
```suggestion
		if (label.color.startsWith('#')) {
			label.color = label.color.replace('#', '');
		}
```

**Review by JakeChampion — COMMENTED** [review]
@luzlab thanks for the pull-request, would you be able to include a unit test for this to confirm it works and to stop future possible regressions?

**Comment by carlo-quinonez** [comment]
Yes, I can work on a unit test (@luzlab is my personal GH Account, this is my work GH account).

Could you help get me started thought? The unit tests are a little hard to follow. I didn't see any obvious place to add a test for parsing the config.

File: bin/github-label-sync.js (hunk 1 above)
Comment (carlo-quinonez) [reply bin/github-label-sync.js:143]: That's what I wrote initially, but then my inner "tester voice" said fewer branches are better LOL! My logic was that no existing colors have the `#` character, so the conditional is unnecessary.

It's trivial either way, I 'll update it as you prefer.

Comment (luzlab) [reply bin/github-label-sync.js]: I added the conditional logic and also convert `color` to a string. I came across a problem with hexadecimal colors like `888888`. Previously, it didn't matter if the colors were a string or number.

**Comment by luzlab** [comment]
@JakeChampion The code for handling the `#` character is in `/bin/github-label-sync.js` and it's not clear how to easily unit test that file. I made the changes you suggested. I could pull the code to normalize `color` into it's own file and write a unit test for just that function. Would that be ok?

**Comment by JakeChampion** [comment]
> @JakeChampion The code for handling the `#` character is in `/bin/github-label-sync.js` and it's not clear how to easily unit test that file. I made the changes you suggested. I could pull the code to normalize `color` into it's own file and write a unit test for just that function. Would that be ok?

If you could do that, that would be fantastic 👍

**Comment by chee** [comment]
@JakeChampion i started doing this work-- but i noticed that if we pull the hash-stripping code out, and then unit test that, it doesn't actually prove that this continues to work (someone could remove the .map(stripLeadingHashesFromColors and nothing would fail).

**Comment by JakeChampion** [comment]
Thanks for looking into this @chee - as the test wouldn't prove it continues to work, shall we merge this as it currently is?

**Comment by chee** [comment]
@JakeChampion i think so! we should maybe open a separate issue to figure out an integration test system for it

**Review by JakeChampion — APPROVED** [review]

**Comment by luzlab** [comment]
Thank you for merging this!

**Comment by danielwerg** [comment]
Shouldn't this be removed from readme?

https://github.com/Financial-Times/github-label-sync/blob/master/README.md?plain=1#L184

**Comment by chee** [comment]
> Shouldn't this be removed from readme?
> 
> https://github.com/Financial-Times/github-label-sync/blob/master/README.md?plain=1#L184

it's not any great emergency because it works with or without the `#`, but if you want to file a PR i'm happy to merge it.

**Comment by danielwerg** [comment]
> > Shouldn't this be removed from readme?
> > https://github.com/Financial-Times/github-label-sync/blob/master/README.md?plain=1#L184
> 
> it's not any great emergency because it works with or without the `#`, but if you want to file a PR i'm happy to merge it.

Submitted a [pr](https://github.com/Financial-Times/github-label-sync/pull/174) some time ago, you can merge it.
